registry/remote/registry_example_test.go

```diff
@@ -0,0 +1,79 @@
+package remote_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"testing"
+
+	"oras.land/oras-go/v2/registry/remote"
+)
+
+func testRegistry() *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		result := struct {
+			Repositories []string `json:"repositories"`
+		}{
+			Repositories: []string{"public/repo1", "public/repo2", "internal/repo3"},
+		}
+		json.NewEncoder(w).Encode(result)
+	}))
+}
+
+var exampleReg *remote.Registry
+
+func TestMain(m *testing.M) {
+	// Mocking local registry
+	ts := testRegistry()
+	defer ts.Close()
+	exampleUri, err := url.Parse(ts.URL)
+	if err != nil {
+		panic(err)
+	}
+	exampleReg, err = remote.NewRegistry(exampleUri.Host) // Create a registry via the remote host
+	if err != nil {
+		panic(err) // Handle error
+	}
+	exampleReg.PlainHTTP = true // Use HTTP
+	os.Exit(m.Run())
+}
+
+// This is a example for listing respositories in the registry with pagination:
```

Comment:
In golang, documentation starts with the function name like
```go
// ExampleRegistry_Repositories is an example for listing respositories in the registry with pagination.
func ExampleRegistry_Repositories()
```
Details see
- https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences
- https://go.dev/doc/effective_go#commentary